doc/src/contracts/multisig_prover.md

```diff
@@ -62,59 +112,113 @@ Prover-->>-Relayer: returns GetProofResponse
+
+
+## Update and confirm WorkerSet graph
 
-#[derive(QueryResponses)]
-pub enum QueryMsg {
-    #[returns(GetProofResponse)]
-    GetProof { multisig_session_id: Uint64 },
-}
+```mermaid
+graph TD
 
-pub enum ProofStatus {
-    Pending,
-    Completed { execute_data: HexBinary }, // encoded data and proof sent to destination gateway
-}
+r[Relayer]
+subgraph Axelar
+b[Prover]
+v[Voting Verifier]
+m[Multisig]
+s[Service Registry]
+end
 
-pub struct GetProofResponse {
-    pub multisig_session_id: Uint64,
-    pub message_ids: Vec<String>,
-    pub data: Data,
-    pub status: ProofStatus,
-}
+r--UpdateWorkerSet-->b
+b--GetActiveWorkers-->s
+b--RegisterWorkerSet-->m
+r--ConfirmWorkerSet-->b
+b--IsWorkerSetConfirmed-->v
 ```
 
-## Events
 
-```Rust
-pub enum Event {
-    ProofUnderConstruction {
-        multisig_session_id: Uint64,
-    },
-    SnapshotRotated {
-        key_id: String,
-        snapshot: Snapshot,
-        pub_keys: HashMap<String, HexBinary>,
-    },
-}
+## Update and confirm WorkerSet sequence diagram
+
+```mermaid
+sequenceDiagram
+autonumber
+participant External Gateway
+participant Relayer
+box LightYellow Axelar
+participant Prover
+participant Service Registry
+participant Voting Verifier
+participant Multisig
+end
+actor Signers
+Relayer->>+Prover: ExecuteMsg::UpdateWorkerSet
+alt no WorkerSet stored
+  Prover->>+Service Registry: QueryMsg::GetActiveWorkers
+  Service Registry-->>-Prover: save new WorkerSet as current WorkerSet
+  Prover->>+Multisig: ExecuteMsg::RegisterWorkerSet
+else existing WorkerSet stored
+  Prover->>+Service Registry: QueryMsg::GetActiveWorkers
+  Service Registry-->>-Prover: save new WorkerSet as next WorkerSet
+  Prover->>+Multisig: ExecuteMsg::StartSigningSession (for operatorship transferred message)
+  Multisig-->>Signers: emit SigningStarted event
+  Multisig->>-Prover: reply with session ID
+  Prover-->>Relayer: emit ProofUnderConstruction event
+  loop Collect signatures
+	  Signers->>+Multisig: signature collection
+  end
+  Multisig-->>-Relayer: emit SigningCompleted event
+  Relayer->>+Prover: QueryMsg::GetProof
+  Prover->>+Multisig: QueryMsg::GetSigningSession
+  Multisig-->>-Prover: reply with status, current signatures vector and snapshot
+  Prover-->>-Relayer: returns GetProofResponse
+  Relayer-->>External Chain: send new WorkerSet to the gateway, signed by old WorkerSet
+  External Chain-->>+Relayer: emit OperatorshipTransferred event
+  Relayer->>+Voting Verifier: ExecuteMsg::ConfirmWorkerSet
+  Voting Verifier-->>-Relayer: emit PollStarted event
+  Relayer->>+Voting Verifier: ExecuteMsg::Vote
+  Relayer->>+Voting Verifier: ExecuteMsg::EndPoll
+  Voting Verifier-->>-Relayer: emit PollEnded event
```

Comment:
this can be omitted. The relayer doesn't care about the PollEnded event, since it ended the poll itself.